cpp/src/arrow/dataset/file_parquet_encryption_test.cc

```diff
@@ -142,12 +145,107 @@ class DatasetEncryptionTest : public ::testing::Test {
       kms_connection_config_;
 };
 
-// This test demonstrates the process of writing a partitioned Parquet file with the same
-// encryption properties applied to each file within the dataset. The encryption
-// properties are determined based on the selected columns. After writing the dataset, the
-// test reads the data back and verifies that it can be successfully decrypted and
-// scanned.
-TEST_F(DatasetEncryptionTest, WriteReadDatasetWithEncryption) {
+class DatasetEncryptionTestUniformEncryption : public ::testing::Test {
+ public:
+  // This function creates a mock file system using the current time point, creates a
+  // directory with the given base directory path, and writes a dataset to it using
+  // provided Parquet file write options. These include an encryption configuration
+  // which uses EncryptionConfiguration.column_keys to specify the encryption key for
+  // each column.
+  // The dataset is partitioned using a Hive partitioning scheme.
+  // The function also checks if the written files exist in the file system.
+  static void SetUpTestSuite() {
```

Comment:
Could you please avoid copy-pasting all this? You should instead create a base class to receive all common setup functionality.